codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -57,15 +58,30 @@ public DefaultHttp2RemoteFlowController(Http2Connection connection) {
         connection.addListener(new Http2ConnectionAdapter() {
             @Override
             public void onStreamAdded(Http2Stream stream) {
-                // Just add a new flow state to the stream.
-                stream.remoteFlowState(new DefaultFlowState(stream, 0));
+                // If the stream state is not open then the stream is not yet eligible for flow controlled frames and
+                // only requires the ReducedFlowState. Otherwise the full amount of memory is required.
+                stream.remoteFlowState(stream.state() == IDLE ?
```

Comment:
@nmittler I agree.

@Scottmitch 1) Why is it relevant what memory usage streams in `idle` state have? 2) Once we switch from `idle` to `open` we have to allocate a real `FlowState` and since `ReducedFlowState + FlowState > FlowState` we will actually use more memory.